subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -1388,32 +1387,25 @@ static inline void ticker_job_worker_bh(struct ticker_instance *instance,
 		ticks_expired += ticks_to_expire;
 
 #if !defined(CONFIG_BT_TICKER_COMPATIBILITY_MODE)
-		if (ticker->lazy_current != 0U &&
-		    !TICKER_RESCHEDULE_PENDING(ticker)) {
+		ticks_latency -= ticks_to_expire;
+#endif /* !CONFIG_BT_TICKER_COMPATIBILITY_MODE */
```

Comment:
I think we need to keep the logic that if lazy_current != 0 and no pending reschedules, then clear the slot_previous info and nothing else. This is needed because must_expire still needs special handling for must_expired nodes that don't go in the air, and catch a lazy count. By checking for this in addition, I think we get the desired functionality for both "worlds". At least my tests pass with this:
```
#if !defined(CONFIG_BT_TICKER_COMPATIBILITY_MODE)
		ticks_latency -= ticks_to_expire;

		if (ticker->must_expire && ticker->lazy_current != 0U &&
		    !TICKER_RESCHEDULE_PENDING(ticker)) {
			instance->ticker_id_slot_previous = TICKER_NULL;
			instance->ticks_slot_previous = 0U;
		} else
#endif /* !CONFIG_BT_TICKER_COMPATIBILITY_MODE */
		{
```

Reply:
A skipped ticker instance shall not clear the `ticks_slot_previous`, wouldn't that also apply to a "must expire" tickers? We should have a audio call at the next opportunity to review this.

Reply:
Yes, you're right. I had a sneaking suspicion after writing that, and went back to the code this morning. I will take a second look, and let's talk about this Monday morning, if you're available.

Reply:
I think I figured it out. The problem is in the updating of slot_previous to the expired ticker id, regardsless of must_expire:
```
	/* If a reschedule is set pending, we will need to keep
	 * the slot_previous information
	 */
	if ((ticker->ticks_slot != 0U) &&
	    !(ticker->must_expire && ticker->lazy_current != 0U) && // <----- added
	    (((ticker->req - ticker->ack) & 0xff) == 2U) &&
	    !TICKER_RESCHEDULE_PENDING(ticker)) {
		instance->ticker_id_slot_previous =
			id_expired;
		instance->ticks_slot_previous =
			ticker->ticks_slot;
	}
```
This will overwrite the info from the blocking node, and use the ticks_slot from the must_expire (shallow) expiry instead, which is incorrect.

Reply:
I have tested and added your check, PR updated.

Reply:
That's great. Doing final re-test here. However, unfortunately you need to apply conditional compile with CONFIG_BT_TICKER_COMPATIBILITY_MODE around that added line :-/

Reply:
my fault, I was aware, but forgot to add the IS_ENABLE(...). Will handle and push updates soon.